services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java

```diff
@@ -608,4 +609,30 @@ public AuthorizationStatus canAccessMembershipInvitationSubmission(UserInfo user
 		}
 		return AuthorizationManagerUtil.accessDenied("Unauthorized to access membership invitation " + mis.getId() + " for " + accessType);
 	}
+
+	@Override
+	public AuthorizationStatus canAccessMembershipInvitationSubmission(String misId, MembershipInvtnSignedToken token, ACCESS_TYPE accessType) {
+		try {
+			SignedTokenUtil.validateToken(token);
+		} catch (IllegalArgumentException e) {
+			return AuthorizationManagerUtil.accessDenied("Unauthorized to access membership invitation " + misId + "(" + e.getMessage() + ")");
+		}
+		if (token.getMembershipInvitationId().equals(misId) && accessType == ACCESS_TYPE.READ) {
+			return AuthorizationManagerUtil.AUTHORIZED;
+		}
+		return AuthorizationManagerUtil.accessDenied("Unauthorized to access membership invitation " + misId + " for " + accessType);
+	}
+
+	@Override
+	public AuthorizationStatus canAccessMembershipInvitationSubmission(Long userId, String misId, InviteeVerificationSignedToken token, ACCESS_TYPE accessType) {

```

Comment:
please omit misId, which is redundant wrt the token content

Reply:
Fixed